Sources/Build/Command.compile(ClangModule).swift

```diff
@@ -66,42 +47,80 @@ private extension ClangModule {
     }
 }
 
-private extension Sources {
-    func compilePathsForBuildDir(_ wd: String) -> [(filename: String, source: String, object: String, deps: String)] {
-        return relativePaths.map { source in
-            let path = Path.join(root, source)
-            let object = Path.join(wd, "\(source).o")
-            let deps = Path.join(wd, "\(source).d")
+/// A helper struct for ClangModule to compute basic
+/// flags needed to compile and link C language targets.
+struct ClangModuleBuildMeta {
```

Comment:
No, not really, and I can't do a survey right now to try and figure out what is better... if you think this is a monotonic improvement I am fine taking it and continuing to refine over time.